README.md

```diff
@@ -1,171 +1,577 @@
+
+## The `EntitySeed<TEntity>` class
+The `EntitySeed<TEntity>` has two important methods:
+
+- `Prerequisites`: get the navigations that should be created as a predecessor to `TEntity`, e.g a required parent.
+
+- `GetDefault`: create an instance of `TEntity` (without prerequisites).
+
+When used together, these methods create a valid instance of `TEntity`.
+
+There are two ways that the library implements the above two methods to seed data.
+
+### Model configuration-based seeding
+> [!IMPORTANT]
+> This is only supported in `Audacia.Seed.EntityFrameworkCore`. For EF6, please see [reflection-based-seeding](#reflection-based-seeding) below.
+
+The library uses EF Core's [`IModel` interface](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.metadata.imodel) to get information about what prerequisites `TEntity` needs to be created - namely, it's required foreign keys. Once we've determined all required foreign keys, we'll construct a `EntitySeed<TNavigation>` for it.
+
+### Reflection-based seeding
 
-        var end = Random.DateTimeFrom(start);
-        
-        return new Holiday
+#### `Prerequisites` 
+When seeding without a class, we figure out the `Prerequisites` as follows:
+1. The library finds all fields that it thinks are navigation properties. This is based on the property being a `class`, and the existance of another property with the name `{PropertyName}Id`.
+2. A navigation property is considered as mandatory if its corresponding foreign key is not nullable.
+3. For each mandatory navigation property, we try to find a `EntitySeed<TNavigation>` class for it. If none is found, we'll repeat this process for `TNavigation`.
+
+#### `GetDefault` 
+When seeding without a class, the library will call the first constructor it finds for `TEntity`, using example values for any constructor arguments as needed:
+- if it is a `string`, a random `Guid` will be used.
+- otherwise, the default value for the type will be used.
+
+### Seeding method for each provider
+The following table shows how each provider behaves when seeding data: 
+
+| Provider | Model configuration | Reflection         |
+| --- |---------------|--------------------|
+| EF Core | ✔️            |  |
+| EF 6 |               | ️✔️️               |
+| In-memory |               | ️️️✔️              |
+
+## Custom `EntitySeed<TEntity>`s
+Creating a child class of `EntitySeed<TEntity>` class allows you define a valid default `TEntity` to be seeded looks like.
+It should seed _exactly_ the information required for an instance of `TEntity` to be considered valid. For example, it may be that you have two optional navigation properties, but your domain ensures exactly one is populated. This is a good use-case for a custom class as the default behaviour will be to seed neither of these navigation properties.
```

Comment:
The example in prose is "two optional navigation properties, but your domain ensures exactly one is populated" but the code example doesn't match that. I found that surprising, could we use the same example in both prose and code? (Or add a second example if we're intentionally trying to show two different scenarios.)

Reply:
sure - changed the example

Reply:
thanks!